### PR TITLE
Make CreateDelegateDto.safe nullable

### PR DESCRIPTION
**Pull request by hectorgomezv**
## Changes
- Make `CreateDelegate.safe` nullable.
- Adjust the rest of the `CreateDelegate` DTO payload members.

### CODE REVIEW

**Comment by hectorgomezv** [comment]
`CreateDelegateSchema` unit tests were added in [999a0aa](https://github.com/safe-global/safe-client-gateway/pull/1275/commits/999a0aa6d988c96e9131cdb559efb9150dcc0ed7)

**Review by iamacook — APPROVED** [review]

File: src/datasources/transaction-api/transaction-api.service.ts [hunk 1]
```diff
@@ -200,11 +200,11 @@ export class TransactionApi implements ITransactionApi {
...
 
   async postDelegate(args: {
-    safeAddress?: string;
-    delegate?: string;
-    delegator?: string;
-    signature?: string;
-    label?: string;
+    safeAddress: string | null;
+    delegate: string;
+    delegator: string;
```
Comment (iamacook) [inline src/datasources/transaction-api/transaction-api.service.ts:205]:
```suggestion
    safeAddress: `0x${string}` | null;
    delegate: `0x${string}`;
    delegator: `0x${string}`;
```

File: src/domain/delegate/delegate.repository.interface.ts [hunk 2]
```diff
@@ -16,11 +16,11 @@ export interface IDelegateRepository {
...
   postDelegate(args: {
     chainId: string;
-    safeAddress?: string;
-    delegate?: string;
-    delegator?: string;
-    signature?: string;
-    label?: string;
+    safeAddress: string | null;
+    delegate: string;
+    delegator: string;
```
Comment (iamacook) [inline src/domain/delegate/delegate.repository.interface.ts:21]:
```suggestion
    safeAddress: `0x${string}` | null;
    delegate: `0x${string}`;
    delegator: `0x${string}`;
```

File: src/domain/delegate/delegate.repository.ts [hunk 3]
```diff
@@ -39,11 +39,11 @@ export class DelegateRepository implements IDelegateRepository {
...
   async postDelegate(args: {
     chainId: string;
-    safeAddress?: string;
-    delegate?: string;
-    delegator?: string;
-    signature?: string;
-    label?: string;
+    safeAddress: string | null;
+    delegate: string;
+    delegator: string;
```
Comment (iamacook) [inline src/domain/delegate/delegate.repository.ts:44]:
```suggestion
    safeAddress: `0x${string}` | null;
    delegate: `0x${string}`;
    delegator: `0x${string}`;
```

File: src/domain/interfaces/transaction-api.interface.ts [hunk 4]
```diff
@@ -42,11 +42,11 @@ export interface ITransactionApi {
...
 
   postDelegate(args: {
-    safeAddress?: string;
-    delegate?: string;
-    delegator?: string;
-    signature?: string;
-    label?: string;
+    safeAddress: string | null;
+    delegate: string;
+    delegator: string;
```
Comment (iamacook) [inline src/domain/interfaces/transaction-api.interface.ts:47]:
```suggestion
    safeAddress: `0x${string}` | null;
    delegate: `0x${string}`;
    delegator: `0x${string}`;
```

File: src/routes/delegates/entities/schemas/create-delegate.dto.schema.ts [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { z } from 'zod';
 import { AddressSchema } from '@/validation/entities/schemas/address.schema';
 
 export const CreateDelegateDtoSchema = z.object({
-  safe: AddressSchema.optional(),
+  safe: AddressSchema.optional().nullable().default(null),
```
Comment (iamacook) [inline src/routes/delegates/entities/schemas/create-delegate.dto.schema.ts:5]:
```suggestion
  safe: AddressSchema.nullish().default(null),
```

**Pull request merged**